src/nvim/api/vim.c

```diff
@@ -487,13 +487,13 @@ Object nvim_exec_lua(String code, Array args, Error *err)
 /// @param log_level  The log level
 /// @param opts       Reserved for future use.
 /// @param[out] err   Error details, if any
-Object nvim_notify(String msg, Integer log_level, Dictionary opts, Error *err)
+Object nvim_notify(String msg, Integer log_level, Dict(empty) *opts, Error *err)
   FUNC_API_SINCE(7)
 {
   MAXSIZE_TEMP_ARRAY(args, 3);
   ADD_C(args, STRING_OBJ(msg));
   ADD_C(args, INTEGER_OBJ(log_level));
-  ADD_C(args, DICTIONARY_OBJ(opts));
+  ADD_C(args, DICTIONARY_OBJ(ARRAY_DICT_INIT));
```

Comment:
This is not backward-compatible.

Reply:
Technically, taking a keydict at all in this case shouldn't be backwards compatible since it eliminates all keys in this case, correct me if I'm wrong.

Reply:
Yes. All changes to `nvim_notify` in this PR should be reverted, otherwise it will break plugins that override `vim.notify()`.

Reply:
It'd break plugins that use `vim.notify` through `nvim_notify` and expect those to be completely equivalent. Is that the case? `nvim_notify`'s docs explicitly say that the `opts` field is `reserved for future use`. If `nvim_notiffy` must always have the same behavior as `vim.notify` than why have it all?

Reply:
IDK. Maybe to make it easier to use from Vimscript?

Reply:
Do you want me to:

a) leave it as is. This will (silently) break code that isn't respecting `nvim_notify`'s API about its `opts` field;
b) revert the changes;
c) revert the changes, and modify `nvim_notify`'s docs to make it clear that the `opts` field has the same exact API as the equivalent field in `vim.notify`.

Reply:
I think it should be c)

Reply:
I've reverted all the changes related to `nvim_notify` in 910da65e49fc186d47c3629a81fede3c17c5b6ca. I'll update the docs in a follow-up PR.